app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt

```diff
@@ -283,6 +238,135 @@ class ReportingRepositoryImpl(
         }
     }
 
+    private suspend fun prepareListOfTemporaryExposureKeysForRedReporting(
+        temporaryExposureKeysFromSDK: List<TemporaryExposureKey>,
+        thresholdTime: Int
+    ): List<TemporaryExposureKeysWrapper> {
+        val temporaryExposureKeysList = mutableListOf<TemporaryExposureKeysWrapper>()
+
+        val sentYellowTemporaryExposureKeys =
+            infectionMessengerRepository.getSentTemporaryExposureKeysByMessageType(
+                MessageType.InfectionLevel.Yellow
+            )
+
+        // In case a yellow reporting happened in the past we report the keys reported yellow and
+        // the newer keys from Exposure SDK.
+        if (sentYellowTemporaryExposureKeys.isNotEmpty()) {
+            temporaryExposureKeysList.addAll(
+                sentYellowTemporaryExposureKeys
+                    .map { message ->
+                        TemporaryExposureKeysWrapper(
+                            message.rollingStartIntervalNumber,
+                            message.password,
+                            MessageType.InfectionLevel.Red
+                        )
+                    }
+            )
+
+            val latestYellowExposureKeys =
```

Comment:
this is not a `key` or `keys` but a timestamp

Reply:
updated